### PR TITLE
feat: support ifgroup level throughput metrics

### PR DESCRIPTION
**Pull request by Hardikl**

### CODE REVIEW

**Review by cgrinds — DISMISSED** [review]

**Comment by Hardikl** [comment]
```
harvest % bin/harvest rest -p '*' show data --api network/ethernet/ports | jq -r '.[] | [.poller, .addr, .num_records, .version, .cluster_name, .poll_ms, .api] |  @csv' | column -ts,
"aff-2510"           10   "9.12.1"  "A250-41-42-43"       50   "network/ethernet/ports"
"Sachin_cluster"      8    "9.15.1"  "bsachin-vsim34"      271  "network/ethernet/ports"
"CG2"                30   "9.15.1"  "Cluster2"            293  "network/ethernet/ports"
"aff-2500"           18   "9.11.0"  "A250-15-28-29"       72   "network/ethernet/ports"
"CG1"                34   "9.15.1"  "Cluster1"            309  "network/ethernet/ports"
"umeng-aff300-05-06" 14   "9.9.1"   "umeng-aff300-05-06"  273  "network/ethernet/ports"
"ontap-tudor2"       34   "9.10.1"  "sg-tme-af200-01-02"  373  "network/ethernet/ports"
"sar"               128  "9.15.1"  "umeng-aff300-01-02"  642  "network/ethernet/ports"
```

```
harvest % bin/harvest rest -p '*' show data --api private/cli/network/port --fields "node,port" | jq -r '.[] | [.poller, .addr, .num_records, .version, .cluster_name, .poll_ms, .api] |  @csv' | column -ts,
"aff-2510"           10   "9.12.1"  "A250-41-42-43"       35   "private/cli/network/port"
"Sachin_cluster"      8    "9.15.1"  "bsachin-vsim34"      283  "private/cli/network/port"
"CG2"                30   "9.15.1"  "Cluster2"            565  "private/cli/network/port"
"aff-2500"           18   "9.11.0"  "A250-15-28-29"       51   "private/cli/network/port"
"CG1"                34   "9.15.1"  "Cluster1"            582  "private/cli/network/port"
"umeng-aff300-05-06" 14   "9.9.1"   "umeng-aff300-05-06"  268  "private/cli/network/port"
"ontap-tudor2"       34   "9.10.1"  "sg-tme-af200-01-02"  315  "private/cli/network/port"
"sar"               128  "9.15.1"  "umeng-aff300-01-02"  835  "private/cli/network/port"
```

```
harvest % bin/harvest rest -p '*' show data --api private/cli/network/port/ifgrp --fields "node,ifgrp" | jq -r '.[] | [.poller, .addr, .num_records, .version, .cluster_name, .poll_ms, .api] |  @csv' | column -ts, 
"aff-2510"           0  "9.12.1"  "A250-41-42-43"       18    "private/cli/network/port/ifgrp"
"Sachin_cluster"     0  "9.15.1"  "bsachin-vsim34"      537   "private/cli/network/port/ifgrp"
"CG2"                0  "9.15.1"  "Cluster2"            528   "private/cli/network/port/ifgrp"
"aff-2500"           0  "9.11.0"  "A250-15-28-29"       1944  "private/cli/network/port/ifgrp"
"CG1"                2  "9.15.1"  "Cluster1"            514   "private/cli/network/port/ifgrp"
"umeng-aff300-05-06" 0  "9.9.1"   "umeng-aff300-05-06"  375   "private/cli/network/port/ifgrp"
"ontap-tudor2"       4  "9.10.1"  "sg-tme-af200-01-02"  545   "private/cli/network/port/ifgrp"
"sar"                2  "9.15.1"  "umeng-aff300-01-02"  527   "private/cli/network/port/ifgrp"
```

As per the details, it doesn't seems 1-1 mapping between call (1 and 2 - they are fine with same key and can add labels appropriately) and 3 (totally different object call).

Hence, we have decided in team internally to remove this rest call and if required later, we can add new template for monitoring the `ifgrp` object separately.

**Comment by Hardikl** [comment]
image

File: cmd/collectors/restperf/plugins/nic/nic.go [hunk 1]
```diff
@@ -133,16 +220,92 @@ func (n *Nic) Run(dataMap map[string]*matrix.Matrix) ([]*matrix.Matrix, *util.Me
...
+		ifgrp := ifgroup.Get("ifgrp").String()
+		ports := ifgroup.Get("ports").Array()
+		for _, portName := range ports {
+			portIfgroupMap[nodeName+portName.String()] = ifgrp
+		}
+	}
+	return portIfgroupMap
+}
+
+func (n *Nic) populateIfgroupMetrics(portIfgroupMap map[string]string, portDataMap map[string]PortData) error {
```
[the remark this answers is not in the document]
Comment (Hardikl) [reply cmd/collectors/restperf/plugins/nic/nic.go]: Done

**Comment by Hardikl** [comment]
image

**Pull request merged**